examples/imagenet/train_imagenet.py

```diff
@@ -271,6 +272,12 @@ def train_loop():
         if train:
             optimizer.zero_grads()
             loss, accuracy = model.forward(x, y)
+            if not graph_generated:
+                from chainer import computational_graph as C
+                with open('graph.dot', 'w') as o:
+                    o.write(str(C.computational_graph((loss,), False)))
+                with open('graph.2.dot', 'w') as o:
+                    o.write(str(C.computational_graph((loss,), True)))
```

Comment:
How about moving this graph generation routine after the `optimizer.update()` line to make the code more understandable?